adaptdl/adaptdl/torch/adascale.py

```diff
@@ -77,10 +77,11 @@ class AdaScale(object):
 
     .. _AdaScale: https://proceedings.icml.cc/static/paper_files/icml/2020/4682-Supplemental.pdf
     """  # noqa: E501
-    def __init__(self, optimizer, scale=None,
+    def __init__(self, optimizer, mp_scaler=None, scale=None,
```

Comment:
With the pluggable scaling rules PR, should figure out a better way to organize this code so that scaling rules don't need to be aware of mixed precision.

Reply:
The issue with having the Adascale code agnostic with mixed precision is that the mixed precision multiplier changes over time and we can only get it from the user's arbitrary `torch.cuda.amp.GradScaler` object. Unless we do something really weird with subclassing/monkeypatching `GradScaler`, the scaling rule will need to be aware of the `GradScaler` object. Alternatively, we could have an `update_amp_scaling` function in `AdaptiveDataParallel`, but I don't think thats any better.